docs/user/building-images.md

```diff
@@ -0,0 +1,268 @@
+
+```console
+OCI_REGISTRY=quay.io
+OCI_IMAGE_REPO=${OCI_REGISTRY}/your_org/centos-bootc-flightctl
+OCI_IMAGE_TAG=v1
+```
+
+Build the OS image for your target platform:
+
+```console
+sudo podman build -t ${OCI_IMAGE_REPO}:${OCI_IMAGE_TAG} .
+```
+
+### Signing and Publishing the OS Image (bootc)
+
+There are several methods for signing container images. We will focus on signing with [Sigstore](https://www.sigstore.dev/) signatures using a private key. For other options, refer to the [RHEL](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/building_running_and_managing_containers/assembly_signing-container-images_building-running-and-managing-containers) or [cosign](https://github.com/sigstore/cosign) documentations.
+
+Generate a Sigstore key pair `signingkey.pub` and `signingkey.private`:
+
+```console
+skopeo generate-sigstore-key --output-prefix signingkey
+```
+
+Configure container tools like Podman and Skopeo to upload Sigstore signatures together with your signed image to your OCI registry:
+
+```console
+sudo tee "/etc/containers/registries.d/${OCI_REGISTRY}.yaml" > /dev/null <<EOF
+docker:
+    ${OCI_REGISTRY}:
+        use-sigstore-attachments: true
+EOF
+```
+
+Log in to your OCI registry, then sign and publish the OS image to it:
+
+```console
+sudo podman login ${OCI_REGISTRY}
+sudo podman push --sign-by-sigstore-private-key ./signingkey.private ${OCI_IMAGE_REPO}:${OCI_IMAGE_TAG}
+```
+
+### Building the OS Disk Image
+
+Next, create a directory called "output" and use `bootc-image-builder` to generate an OS disk image of type "raw" from your OS image:
+
+```console
+mkdir -p output
+
+sudo podman run --rm -it --privileged --pull=newer \
+    --security-opt label=type:unconfined_t \
+    -v $(pwd)/output:/output \
+    -v /var/lib/containers/storage:/var/lib/containers/storage \
+    quay.io/centos-bootc/bootc-image-builder:latest \
+    --type raw \
+    ${OCI_IMAGE_REPO}:${OCI_IMAGE_TAG}
+```
+
+Once `bootc-image-builder` completes, you can find the disk image under `$(pwd)/output/image/disk.raw`.
+
+Refer to `bootc-image-builder`'s [list of image types](https://github.com/osbuild/bootc-image-builder?tab=readme-ov-file#-image-types) for other supported types.
+
+### Signing and Publishing the OS Disk Image (bootc)
+
+Optionally, you can sign and publish your disk image to your OCI registry, too. This helps unify hosting and distribution. Using manifest lists, you can even keep matching bootc and disk images together:
+
+```console
+sudo podman manifest create ${OCI_IMAGE_REPO}-unified:${OCI_IMAGE_TAG}
+sudo podman manifest add \
+    ${OCI_IMAGE_REPO}-unified:${OCI_IMAGE_TAG} docker://${OCI_IMAGE_REPO}:${OCI_IMAGE_TAG}
+sudo podman manifest add \
+    --artifact --artifact-type application/x-raw-disk-image \
+    --os=linux --arch=x86_64 --annotation disktype=raw \
+    ${OCI_IMAGE_REPO}-unified:${OCI_IMAGE_TAG} $(pwd)/output/image/disk.raw
+sudo podman manifest push --all \
```

Comment:
Need to add the image signing `--sign-by-sigstore-private-key ./signingkey.private`.